src/isomorphic/modern/element/__tests__/ReactJSXElementValidator-test.js

```diff
@@ -99,19 +100,49 @@ describe('ReactJSXElementValidator', () => {
       },
     };
 
-    void <Component>{iterable}</Component>;
+    ReactTestUtils.renderIntoDocument(<Component>{iterable}</Component>);
 
     expectDev(console.error.calls.count()).toBe(1);
     expectDev(console.error.calls.argsFor(0)[0]).toContain(
       'Each child in an array or iterator should have a unique "key" prop.',
     );
   });
 
+  it('does not warns for fragments of multiple elements without keys', () => {
+    spyOn(console, 'error');
+
+    ReactTestUtils.renderIntoDocument(
+      <React.Fragment>
+        <span>1</span>
+        <span>2</span>
+      </React.Fragment>,
+    );
+
+    expectDev(console.error.calls.count()).toBe(0);
+  });
+
+  it('warns for fragments of multiple elements with same key', () => {
+    spyOn(console, 'error');
+
+    ReactTestUtils.renderIntoDocument(
+      <React.Fragment>
+        <span key="a">1</span>
+        <span key="a">2</span>
+        <span key="b">3</span>
+      </React.Fragment>,
+    );
+
+    expectDev(console.error.calls.count()).toBe(1);
+    expectDev(console.error.calls.argsFor(0)[0]).toContain(
+      'Encountered two children with the same key',
```

Comment:
Worth including the key into the message? To make sure we don't mess it up when printing.